modules/outbound-command-event-handler/test/integration/application/process_sdk_outbound_bulk_quotes_request.test.ts

```diff
@@ -0,0 +1,327 @@
+            transactionId: randomGeneratedTransferIds[1],
+            acceptParty: true
+          }
+        ]
+      },
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processSDKOutboundBulkAcceptPartyInfoCommandEventObj = new ProcessSDKOutboundBulkAcceptPartyInfoCmdEvt(
+      processSDKOutboundBulkAcceptPartyInfoCommandEventData
+    );
+    await producer.sendCommandEvent(processSDKOutboundBulkAcceptPartyInfoCommandEventObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Check that the global state to be DISCOVERY_ACCEPTANCE_COMPLETED
+    const bulkStateTwo = await bulkTransactionEntityRepo.load(bulkTransactionId);
+    expect(bulkStateTwo.state).toBe(BulkTransactionInternalState.DISCOVERY_ACCEPTANCE_COMPLETED);
+
+    // Check that individual transfers have been updated to DISCOVERY_ACCEPTED
+    const afterIndividualTransfer1 = await bulkTransactionEntityRepo.getIndividualTransfer(bulkTransactionId,  randomGeneratedTransferIds[0]);
+    expect(afterIndividualTransfer1.state).toBe(IndividualTransferInternalState.DISCOVERY_ACCEPTED);
+
+    const afterIndividualTransfer2 = await bulkTransactionEntityRepo.getIndividualTransfer(bulkTransactionId,  randomGeneratedTransferIds[1]);
+    expect(afterIndividualTransfer2.state).toBe(IndividualTransferInternalState.DISCOVERY_ACCEPTED);
+
+    // Check that command handler sends event to domain handler
+    const hasSDKOutboundBulkAcceptPartyInfoProcessed = (domainEvents.find((e) => e.getName() === 'SDKOutboundBulkAcceptPartyInfoProcessedDmEvt'));
+    expect(hasSDKOutboundBulkAcceptPartyInfoProcessed).toBeTruthy();
+
+    // Simulate domain handler sending command event for sdk outbound bulk quotes request
+    const processSDKOutboundBulkQuotesRequestCommandEventData : IProcessSDKOutboundBulkQuotesRequestCmdEvtData = {
+      bulkId: bulkTransactionId,
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processSDKOutboundBulkQuotesRequestCommandEventObj = new ProcessSDKOutboundBulkQuotesRequestCmdEvt(
+      processSDKOutboundBulkQuotesRequestCommandEventData
+    );
+    await producer.sendCommandEvent(processSDKOutboundBulkQuotesRequestCommandEventObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Check that the global state of bulk to be AGREEMENT_PROCESSING
+    const bulkStateThree = await bulkTransactionEntityRepo.load(bulkTransactionId);
+    expect(bulkStateThree.state).toBe(BulkTransactionInternalState.AGREEMENT_PROCESSING);
+
+    // Check that command handler published BulkQuotesRequested message
+    const hasBulkQuotesRequested = (domainEvents.find((e) => e.getName() === 'BulkQuotesRequestedDmEvt'));
+    expect(hasBulkQuotesRequested).toBeTruthy();
+
+    // Check that bulk batches have been created.
+    const bulkBatchIds = await bulkTransactionEntityRepo.getAllBulkBatchIds(bulkTransactionId);
+    expect(bulkBatchIds[0]).toBeDefined();
```

Comment:
Can we also check the state for these batches in redis?

Reply:
It will be very valuable if we can test for a receiver id, DISCOVERY_ACCEPTED, maxConfigPerBatch combination and having multiple batch events being created for different combinations.

Reply:
done. added assert for AGREEMENT_PROCESSING state.